kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/WrapperVersion.java

```diff
@@ -29,9 +29,34 @@
  * The version of the wrapper schema used to persist information in the wrapper.
  */
 public enum WrapperVersion {
+    V1_UNSUPPORTED {
+        @Override
+        public <E> void writeWrapper(@NonNull Serde<E> edekSerde, @NonNull E edek, @NonNull String topicName, int partitionId, @NonNull RecordBatch batch,
+                                     @NonNull Record kafkaRecord, @NonNull Dek<E>.Encryptor encryptor, @NonNull ParcelVersion parcelVersion, @NonNull AadSpec aadSpec,
+                                     @NonNull Set<RecordField> recordFields, @NonNull ByteBuffer buffer) {
+            throw unsupportedVersionException();
+        }
+
+        @Override
+        public <E> void read(@NonNull ParcelVersion parcelVersion, @NonNull String topicName, int partition, @NonNull RecordBatch batch, @NonNull Record record,
+                             ByteBuffer wrapper, Dek<E>.Decryptor decryptor, @NonNull BiConsumer<ByteBuffer, Header[]> consumer) {
+            throw unsupportedVersionException();
+        }
+
+        @Override
+        public <E, T> T readSpecAndEdek(ByteBuffer wrapper, Serde<E> serde, BiFunction<CipherSpec, E, T> fn) {
+            throw unsupportedVersionException();
+        }
+
+        @NonNull
+        private static EncryptionException unsupportedVersionException() {
+            return new EncryptionException("unsupported wrapper version");
```

Comment:
Tempted to say this should be single field for performance reasons, given it could occur thousands of times trying to replay a problematic topic. 

Is unsupported wrapper version wrapper version going to mean anything to an operator reading the logs? I think we should probably include the version in the exception. 

```suggestion
            return new EncryptionException("V1 wrappers are unsupported");
```

Reply:
It shouldn't get this far because during decrypt we would throw when we encounter `kroxylicious.io/encryption: 1`   in the header with error message `Deserialization of EncryptionVersion=1 records is not supported by this version of the encryption filter.`

I'll make the message more specific though